frontend/src/views/QCAllotmentForm.vue

```diff
@@ -0,0 +1,157 @@
+      <v-autocomplete
+        v-model="allotment.assignee"
+        return-object
+        :items="users || []"
+        item-text="name"
+        label="Select an assignee"
+        clearable
+        dense
+      >
+        <template slot="item" slot-scope="{ item }">
+          <template v-if="typeof item !== 'object'">
+            <v-list-tile-content v-text="item"></v-list-tile-content>
+          </template>
+          <template v-else>
+            <v-list-tile-content>
+              <v-list-tile-title v-html="item.name"></v-list-tile-title>
+              <v-list-tile-sub-title v-html="item.emailAddress"></v-list-tile-sub-title>
+            </v-list-tile-content>
+          </template>
+        </template>
+      </v-autocomplete>
+      <v-layout align-center class="pa-2">
+        <v-flex>
+          <code>{{ task[".key"] }}</code>
+          <sound-quality-badge :task="task"></sound-quality-badge>
+        </v-flex>
+        <v-flex>
+          <sound-issues-list :item="task"></sound-issues-list>
+        </v-flex>
+        <v-flex>
+          <a :href="originalFile" target="_blank" rel="noopener noreferrer" v-if="originalFile">Original file</a>
+          <span v-else>
+            No original file
+          </span>
+          <span> | </span>
+          <a :href="restoredFile" target="_blank" rel="noopener noreferrer" v-if="restoredFile">Restored file</a>
+          <span v-else>
+            No restored file
+          </span>
+        </v-flex>
+      </v-layout>
+      <v-textarea box label="Comment" v-model="allotment.comment">
+      </v-textarea>
+      <v-btn @click="submit">submit</v-btn>
+    </v-form>
+    <div v-else>
+      <div v-if="submissionStatus === 'complete'">Success</div>
+      <div v-else>invalid task</div>
+    </div>
+  </div>
+</template>
+
+<script lang="ts">
+import { Component, Mixins } from "vue-property-decorator";
+import UsersByRole from "@/mixins/UsersByRole";
+import firebase from "firebase";
+import fb from "@/firebaseApp";
+import SoundQualityBadge from "@/components/SoundQualityBadge.vue";
+import SoundIssuesList from "@/components/SE/SoundIssuesList.vue";
+import { getListId } from "@/utility";
+
+export const storage = fb.storage();
+
+@Component({
+  name: "QCAllotmentForm",
+  components: { SoundIssuesList, SoundQualityBadge }
+})
+export default class QCAllotmentForm extends Mixins<UsersByRole>(UsersByRole) {
+  usersRole = "QC";
+  allotment: any = {};
+  submissionStatus: any = null;
+  task: any = null;
+  originalFile: any = null;
+  restoredFile: any = null;
+
+  mounted() {
+    this.getData();
+  }
+
+  get taskId() {
+    return this.$route.params.taskId;
+  }
+
+  getTask() {
+    const listId = getListId(this.taskId);
+    this.$bindAsObject(
+      "task",
+      fb.database().ref(`sound-editing/tasks/${listId}/${this.taskId}`)
+    );
+  }
+
+  baseFilePath() {
+    return process.env.VUE_APP_STORAGE_ROOT_DOMAIN;
```

Comment:
I don't think it's worth having this as a separate method. Firstly, this is not a base file path but rather it is a base domain name. Secondly, using `process.env.VUE_APP_STORAGE_ROOT_DOMAIN` in the code is self-explaining and not much longer. So I propose that we eliminate this method.